improver/utilities/spatial.py

```diff
@@ -160,10 +160,304 @@ def number_of_grid_cells_to_distance(cube: Cube, grid_points: int) -> float:
+        """
+        lat_diffs = np.diff(self.lats)
+
+        y_distances = self.sphere_radius * np.deg2rad(lat_diffs)
+
+        y_distances_grid = np.tile(np.expand_dims(y_distances, axis=1), len(self.longs))
+        dims = [(self.y_separation_axis, 0), (self.cube.coord(axis="x"), 1)]
+        return self.build_distances_cube(y_distances_grid, dims, "y")
+
+
+class ProjectionCubeDistanceCalculator(BaseDistanceCalculator):
+    def __init__(self, cube: Cube):
+        """
+        Args:
+            cube:
+                Cube for which the distances will be calculated.
+        """
+        if cube.coord(axis="x").circular:
+            raise NotImplementedError(
+                "Cannot calculate distances between bounding points of a circular projected "
+                "coordinate."
+            )
+        super().__init__(cube)
+
+    def _get_x_distances(self) -> Cube:
+        """
+        Calculates the x-axis distances between adjacent grid points of a cube which uses
+        Equal Area coordinates.
+
+        Returns:
+            A cube containing the x-axis distances between the grid points of the input
+            cube in metres.
+        """
+        x_distances = calculate_grid_spacing(self.cube, axis="x", units="m")
+        data = np.full(
+            (self.cube.shape[0], len(self.x_separations_axis.points)), x_distances
+        )
+        dims = [
+            (self.cube.coord("projection_y_coordinate"), 0),
+            (self.x_separations_axis, 1),
+        ]
+        return self.build_distances_cube(data, dims, "x")
+
+    def _get_y_distances(self) -> Cube:
+        """
+        Calculates the y-axis distances between adjacent grid points of a cube which uses
+        Equal Area coordinates.
+
+        Returns:
+            A cube containing the vertical distances between the grid points of the input
+            cube in metres.
+        """
+        y_grid_spacing = calculate_grid_spacing(self.cube, axis="y", units="m")
+        data = np.full(
+            (len(self.y_separation_axis.points), self.cube.data.shape[1]),
+            y_grid_spacing,
+        )
+        dims = [
+            (self.y_separation_axis, 0),
+            (self.cube.coord("projection_x_coordinate"), 1),
+        ]
+        return self.build_distances_cube(data, dims, "y")
+
+
+class DistanceBetweenGridSquares(BasePlugin):
+    """
+    Calculates the distances between adjacent grid squares within a cube.
+    The distances are calculated along the x and y axes individually.
+    Returned distances are in metres.
+    The class can handle cubes with either Geographic (lat-long) or Equal Area projections.
+    For lat-lon cubes, the distances are calculated assuming a spherical earth.
+    This causes a < 0.15% error compared with the full haversine formula.
+    """
+
+    def __init__(self, cube: Cube):
+        """
+        Args:
+            cube:
+                Cube for which the distances will be calculated.
```

Comment:
I think it is because it is used to determine which child class to use. I think this discrepancy could be problematic and it breaks the convention for an IMPROVER plugin, so I've changed it.